apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.html.ts

```diff
@@ -0,0 +1,340 @@
+import { html, joinHtml, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+
+import { compiledScriptTag, nodeModulesAssetPath } from '../../lib/assets.js';
+import { config } from '../../lib/config.js';
+
+export function InstructorFileEditor({ resLocals }: { resLocals: Record<string, any> }) {
+  const { fileEdit, authz_data, course_owners, course, __csrf_token } = resLocals;
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${renderEjs(import.meta.url, "<%- include('../partials/head'); %>", {
+          ...resLocals,
+          pageTitle: `Edit ${fileEdit?.fileName}`,
+        })}
+        <!-- This is typically imported in the compiled asset, but the way ace handles modes/themes by dynamically loading scripts requires it to be loaded directly from ace-builds -->
+        <script src="${nodeModulesAssetPath('ace-builds/src-min-noconflict/ace.js')}"></script>
```

Comment:
@nwalters512 if you have a better way to handle this, that would be great. In essence, when you set the mode and theme in ace, it dynamically loads the mode/theme script tags, using the location of the ace object as a basis. If I remove the tag above, the location is based on the asset prefix, which is incorrect. If I keep the tag above it works, but the ace codebase is duplicated, since the client also imports it. I would imagine that some way to override the mode/theme loader/require in ace may allow this to be handled, but I'm not sure how this could take place.

Reply:
Found a way to handle this in 0053a318d.